[FIX] html_builder, website: reordering issue in gallery slideshow

shsa-odoo opened the pull request:
- Enhanced the image gallery plugin to retain inline styles and
    attributes of the active image when re-rendering the slideshow.
    This ensures visual modifications are not lost during transitions.
- Updated the related website tour (`snippet_image_gallery`) to use new
    popover-based option selectors and added assertions to validate
    style persistence after navigation.

robodoo commented:
This PR targets the un-managed branch odoo-dev/odoo:master-mysterious-egg, it needs to be retargeted before it can be merged.

robodoo commented:
This PR targets the un-managed branch odoo-dev/odoo:master-mysterious-egg, it needs to be retargeted before it can be merged.

msh-odoo commented on addons/html_builder/static/src/website_builder/plugins/options/image_gallery_option_plugin.js:
```diff
@@ -317,8 +317,21 @@ class ImageGalleryOption extends Plugin {
             carouselEl.removeEventListener("slid.bs.carousel", this.onCarouselSlid);
         }
         container.replaceChildren(slideshowEl);
+        const modifiedImage = images.find((img) =>
```
What if I apply multiple options on multiple images and then I re-order the images, the current ```.find1``` function will only finds first image, I guess this is wrong, don't you think?

shsa-odoo replied:
Correct!! the current .find() function will only retrieve the first image in the images array that has the o_modified_image_to_save class. If multiple images have this class (e.g., when multiple options are applied to multiple images), the current logic will not address them all.
Thanks for the review will check it and update for that case.

msh-odoo commented:
@ged-odoo @FrancoisGe Ready for review, would you please review an merge?

msh-odoo commented:
@ged-odoo @FrancoisGe Ready for review and merge.

Jinjiu96 commented:
Hello, thanks for the fix! If I understood correctly, the pr is to fix lost attributes when reordering a carousel like image gallery. For this issue, we already have a merged fix: https://github.com/odoo-dev/odoo/pull/4522

Could you rebase on the latest egg branch and see if the issue is solved? If so, we can remove the fix :pray:

shsa-odoo commented:
Hello, @Jinjiu96 I have checked that this issue is resolved in latest code, I have removed my fix and updated the PR with tour related changes only. You can review and merge it Thanks!
